distributed/tests/test_worker.py

```diff
@@ -237,6 +235,20 @@ def f():
         assert w.port > 1024
 
     loop.run_sync(f)
+
+
+def test_worker_waits_for_center_to_come_up(loop):
+    @gen.coroutine
+    def f():
+        w = Worker('127.0.0.1', 8007, ip='127.0.0.1')
+        yield w._start()
+
+    try:
+        loop.run_sync(f, timeout=5)
+    except TimeoutError:
+        pass
```

Comment:
Is there a way to make this test shorter?  Five seconds is a non-trivial wait time for a single test.  Alternatively, we could mark this test as `slow` with the `@pytest.mark.slow` decorator.